tests/units/DB.php

```diff
@@ -856,4 +856,151 @@ public function testDecodeFetchResult(string $method, mixed $row, mixed $expecte
         $this->newTestedInstance();
         $this->variable($this->testedInstance->{$method}($mysqli_result))->isEqualTo($expected);
     }
+
+    protected function testGetFieldPatternProvider(): array
+    {
+        return [
+            [
+                'params' => [
+                    'table' => 'glpi_computers',
+                    'field' => 'name', //string
+                    'with_delimiter' => true
+                ],
+                'expected' => [
+                    '/^(NULL|null|(\^|>|<|>=|<=|%)?.*(%|\$)?)$/',
+                    ''
+                ]
+            ],
+            [
+                'params' => [
+                    'table' => 'glpi_computers',
+                    'field' => 'id', //unsigned integer
+                    'with_delimiter' => true
+                ],
+                'expected' => [
+                    '/^(NULL|null|(\^|>|<|>=|<=|%)?\d+(%|\$)?)$/',
+                    'must be a positive integer'
+                ]
+            ],
+            [
+                'params' => [
+                    'table' => 'glpi_entities',
+                    'field' => 'cartridges_alert_repeat', //integer
+                    'with_delimiter' => true
+                ],
+                'expected' => [
+                    '/^(NULL|null|(\^|>|<|>=|<=|%)?-?\d+(%|\$)?)$/',
+                    'must be an integer'
+                ]
+            ],
+            [
+                'params' => [
+                    'table' => 'glpi_budgets',
+                    'field' => 'value', //decimal
+                    'with_delimiter' => true
+                ],
+                'expected' => [
+                    '/^(NULL|null|(\^|>|<|>=|<=|%)?\d+(\.\d+)?(%|\$)?)$/',
+                    'must be a decimal number'
+                ]
+            ],
+            [
+                'params' => [
+                    'table' => 'glpi_computermodels',
+                    'field' => 'depth', //float
+                    'with_delimiter' => true
+                ],
+                'expected' => [
+                    '/^(NULL|null|(\^|>|<|>=|<=|%)?\d+(\.\d+)?(%|\$)?)$/',
+                    'must be a float number'
+                ]
+            ],
+            [
+                'params' => [
+                    'table' => 'glpi_tickets',
+                    'field' => 'solvedate', //timestamp
+                    'with_delimiter' => true
+                ],
+                'expected' => [
+                    '/^(NULL|null|(\^|>|<|>=|<=|%)?(\d{0,4}-?\d{0,2}-?\d{0,2} ?\d{0,2}:?\d{0,2}:?\d{0,2})(%|\$)?)$/',
```

Comment:
This will allow `20230701140005` format for `2023-07-01 14:00:05` datetime. Should we allow this?

Also, it does not seems to allow searching on partial date string. For instance, it is currently possible to search for  `2023`, `2023-07`, `2023-06-2` (from day 20 to 29), ... We should probably keep this feature.

Reply:
> Also, it does not seems to allow searching on partial date string. For instance, it is currently possible to search for `2023`, `2023-07`, `2023-06-2` (from day 20 to 29), ... We should probably keep this feature.

It's still possible

![image](https://github.com/glpi-project/glpi/assets/8530352/f42c8262-da37-4f9e-8d86-9fcb346e3bcc)